Updating CONTRIBUTING.md for eclipse/che#21363 (#2239)
che-incubator/chectl: CONTRIBUTING.md

The development flow includes:

- [Contributing](#contributing)
  - [Start Workspace](#start-workspace)
  - [Build](#build)
  - [Run tests](#run-tests)
  - [Run](#run)
  - [Package binaries](#package-binaries)
  - [Push changes, provide Pull Request](#push-changes-provide-pull-request)
  - [Make Release](#make-release)

## Start Workspace

> To start a workpace from the [devfile](devfile.yaml) create a workspace URL by appending the chectl repo to the URL of your Che host. For example, to create a workspace in the openshift sandbox use the URL:

Reviewer: ```suggestion
To start a workpace from the [devfile](devfile.yaml), create a workspace URL by appending the chectl repository to the URL of your Che host. For example, to create a workspace in the OpenShift sandbox, use the following URL:
```

Reviewer 2: FYI there is a typo on workpace --> workSpace
```suggestion
To start a workspace from the [devfile](devfile.yaml) create a workspace URL by appending the chectl repo to the URL of your Che host. For example, to create a workspace in the openshift sandbox use the URL:
```

Reviewer 3: Probably want to capitalize `Sandbox` as seen on https://developers.redhat.com/developer-sandbox

```
https://devspaces.apps.sandbox-m2.ll9k.p1.openshiftapps.com/f?url=https://github.com/che-incubator/chectl
```

> See more about [devfile](https://devfile.io/docs/2.1.0/what-is-a-devfile)
> See more about [workspace URLs](https://www.eclipse.org/che/docs/stable/end-user-guide/starting-a-new-workspace-with-a-clone-of-a-git-repository/)

> After starting the workspace Theia will clone sources of `chectl` to `/projects/chectl` directory.

Reviewer: ```suggestion
After starting the workspace, Theia will clone sources of `chectl` to the `/projects/chectl` directory.
```

Reviewer 3: Since we will soon be switching IDE default from Theia to VSCode, perhaps "After starting the workspace, the IDE will clone sources..." is better?

> `chectl` is written in TypeScript. For its development there is a dedicated `dev` container which has preinstalled software for comfortable development. That allows the developer to build, test and launch `chectl` inside the container.

Reviewer: ```suggestion
`chectl` is written in TypeScript. For its development, there is a dedicated `dev` container that has preinstalled software for comfortable development. This allows the developer to build, test, and launch `chectl` inside the container.
```

Your workspace will be initialized with a list of commands described in the [devfile](devfile.yaml) `commands` section. The commands are:
- build
- test
- run
- package-binaries
- format-sources

> You can run commands through the Terminal menu by clicking `Terminal => Run Task...` then selecting the desired task. 

Reviewer: ```suggestion
You can run thee commands through the **Terminal** menu by clicking **Terminal > Run Task** and selecting the desired task. 
```

## Build

```bash
yarn
```

> Running the  `che: build` command will run `yarn` in the `/projects/chectl` directory inside `dev` container. The command will install all necessary dependencies and compile the project. Upon successfull assembly, a new `bin` directory will appear in project directory and will have two files: `run` and `run.cmd`.

Reviewer: ```suggestion
Running the  `che: build` command will run `yarn` in the `/projects/chectl` directory inside `dev` container. The command will install all necessary dependencies and compile the project. Upon successful assembly, a new `bin` directory will appear in the project directory and will have two files: `run` and `run.cmd`.
```

yarn test
```

Tests for `chectl` are written by using the [jest](https://jestjs.io/docs/getting-started) framework.

> To run tests, find and launch `che: test` command. The command will run `yarn test` inside `dev` container.

Reviewer 4: ```suggestion
To run tests, find and launch the `che: test` command. The command will run `yarn test` inside `dev` container.
```

```

> Tests for `chectl` are written using `jest` framework.

Reviewer: ```suggestion
Tests for `chectl` are written by using the [jest](https://jestjs.io/docs/getting-started) framework.
```

./bin/run --help
```

To test ensure `chectl` is built successfully, launch the `che: run` command. It wil run `chectl` with `--help` directive.

## Package binaries

> For packaging binaries, [oclif](https://github.com/oclif/dev-cli) is used. It generates packages for Linux, Windows and MacOS operation systems and puts the result in the `dist/channels/stable` directory.

Reviewer: ```suggestion
For packaging binaries, [oclif](https://github.com/oclif/dev-cli) is used. It generates packages for Linux, Windows, and MacOS operation systems and puts the result in the `dist/channels/stable` directory.
```

Reviewer 3: "operation systems" => "operating systems"

> For packaging binaries, [oclif](https://github.com/oclif/dev-cli) is used. It generates packages for Linux, Windows, and MacOS operation systems and puts the result in the `dist/channels/stable` directory.

Reviewer 5: ```suggestion
For packaging binaries, [oclif](https://github.com/oclif/dev-cli) is used. It generates packages for Linux, Windows, and MacOS operating systems and puts the result in the `dist/channels/stable` directory.
```

> To start packaging just run the `che: package-binaries` command. It will run the following in `/projects/chectl` directory:

Reviewer: ```suggestion
To start packaging, just run the `che: package-binaries` command. It will run the following in `/projects/chectl` directory:
```

> To start packaging, just run the `che: package-binaries` command. It will run the following in `/projects/chectl` directory:

Reviewer 4: ```suggestion
To start packaging, just run the `che: package-binaries` command. It will run the following in the `/projects/chectl` directory:
```

## Push changes, provide Pull Request

`chectl` uses several Pull Request checks:
 - The [Conventional Commits](https://conventionalcommits.org) convention for the commit messages.

> There is a required pull request check named **Semantic Pull Request** that ensures that all commits messages are correctly set up. In order to merge a pull request, it has to be green.

Reviewer 4: ```suggestion
There is a required pull request check named **Semantic Pull Request** that ensures that all commit messages are correctly set up. In order to merge a pull request, it has to be green.
```

$ ./run-script-in-docker.sh make-release.sh 7.0.0
```

> Commit the changes of the script and then push to the release branch by overriding current remote release branch.

Reviewer 4: ```suggestion
Commit the changes of the script and then push to the release branch by overriding the current remote release branch.
```
